patroni/postgresql/slots.py

```diff
@@ -47,65 +95,178 @@ def drop_replication_slot(self, name):
         # In normal situation rowcount should be 1, otherwise either slot doesn't exists or it is still active
         return cursor.rowcount == 1
 
-    def sync_replication_slots(self, cluster):
+    def _drop_incorrect_slots(self, cluster, slots):
+        # drop old replication slots which are not presented in desired slots
+        for name in set(self._replication_slots) - set(slots):
+            if not self.ignore_replication_slot(cluster, name) and not self.drop_replication_slot(name):
```

Comment:
if Patroni attempts and fails to delete a slot, can the slot be automatically considered unsafe to use ?